### PR TITLE
Remove `unknown` types

### PR DESCRIPTION
**Pull request by dmfalke**
This PR removes `unknown` types from `Mesa` type definitions.

A big change is that `MesaColumn` now takes two generic types: `Row` and `Key`. The `Row` type is required, and the default for `Key` is `keyof Row`. This covers most use cases, and allows us to better infer the type of `value` in `CellProps`.

Getting these types right is tricky, because `Mesa` tries to allow any data structure to be used for a row, and to allow the consumer to compute the rendered value however it sees fit.

In most cases, we simply use objects as rows, and the keys of those objects as column keys, which this PR attempts to make easy.

### CODE REVIEW

File: packages/libs/coreui/src/components/Mesa/types.ts [hunk 1]
```diff
@@ -35,33 +35,48 @@ export interface MesaStateProps<Row, Key extends string> {
...
 
-interface MesaAction {
+interface MesaAction<Row, Key> {
   selectionRequired?: boolean;
   element: React.ReactNode;
-  callback?: unknown;
-  handler?: unknown;
+  callback?: (row: Row, columns: MesaColumn<Row, Key>[]) => void;
+  handler?: (
+    seection: Row[],
```
Comment (jernestmyers) [inline packages/libs/coreui/src/components/Mesa/types.ts:64]: Nit: typo!

File: packages/libs/coreui/src/components/Mesa/types.ts [hunk 2]
```diff
@@ -35,33 +35,48 @@ export interface MesaStateProps<Row, Key extends string> {
     selectedNoun?: string;
     selectedPluralNoun?: string;
     searchPlaceholder?: string;
-    deriveRowClassName?: unknown; // a function
-    renderEmptyState?: unknown; // a function
-    isRowSelected?: unknown; // a callback function
+    deriveRowClassName?: (row: Row) => string | undefined; // a function
+    renderEmptyState?: () => ReactNode; // a function
+    isRowSelected?: (row: Row) => boolean; // a callback function
```
Comment (jernestmyers) [inline packages/libs/coreui/src/components/Mesa/types.ts:40]: Can we remove these comments here and elsewhere now that the types are more clear?

**Review by jernestmyers — APPROVED** [review]
This looks great and compiled nicely when spinning up a local `genomics-site` and `clinepi-site` dev server.

I left just a couple nits.

**Comment by dmfalke** [comment]
Thanks for the review, @jernestmyers. I've addressed your feedback.

**Pull request merged**